src/renderers/shared/reconciler/__tests__/ReactCompositeComponent-test.js

```diff
@@ -783,7 +812,7 @@ describe('ReactCompositeComponent', function() {
 
       render: function() {
         var output = <Child />;
-        if (!this.state.flag) {
+        if (this.state.flag) {
```

Comment:
I wanted to see why the test was throwing the warning 
1. in the original test we rendered `<Parent> <Child /> <Parent>`, and on `setState` we change the render to `<Parent><span>Child</span></Parent>`
2. in the changed test which doesn't create a warning we go from `<P><span>Child</span> </P>` to `<P><Child /><P>`